pkg/reconciler/managed/reconciler.go

```diff
@@ -412,6 +437,16 @@ func WithPollInterval(after time.Duration) ReconcilerOption {
 	}
 }
 
+// WithCreationGracePeriod configures an optional period during which we will
+// wait for the external API to report that a newly created external resource
+// exists. This allows us to tolerate eventually consistent APIs that do not
+// immediately report that newly created resources exist when queried.
+func WithCreationGracePeriod(d time.Duration) ReconcilerOption {
+	return func(r *Reconciler) {
+		r.creationGracePeriod = d
```

Comment:
What do you think about giving this a default like 10 seconds? I think that'd cover most (all?) cases we want to cover without having to think about which resources need it and update their code.

Reply:
I went with 30.